### PR TITLE
Revert "feat(std/node/fs): adding readdir, rename, and some others (#…

### PR DESCRIPTION
**Pull request by bartlomieju**
…7666)"

This reverts commit 40324ff74816a99ea061929ece1c6a4ff3078bc3.

This commit caused `std_tests` to silently exit process because of forever-pending promise (bug reported in https://github.com/denoland/deno/issues/7088)

CC @AliBasicCoder

### CODE REVIEW

**Review by ry — APPROVED** [review]
We have discovered the std_tests can have silent failures. This PR was subsequently discovered to have failures.

Please resubmit once we correct the underlying issue.

**Comment by AliBasicCoder** [comment]
> We have discovered the std_tests can have silent failures. This PR was subsequently discovered to have failures.
> 
> Please resubmit once we correct the underlying issue.

the fail tests are:
- watching a file
- ASYNC: removing non-empty folder
- SYNC: removing non-empty folder

the fix for the first is adding `return` at the end of the test
the second and third tests cause this:
```
AssertionError: Test case is leaking resources.
Before: {
  "0": "stdin",
  "1": "stdout",
  "2": "stderr"
}
After: {
  "0": "stdin",
  "1": "stdout",
  "2": "stderr",
  "9": "fsFile",
  "10": "fsFile",
  "11": "fsFile"
}

Make sure to close all open resource handles returned from Deno APIs before
```

even every file or folder created is deleted after the test

**Comment by AliBasicCoder** [comment]
NOTE: all of my test did resolve or reject

File: std/node/_fs/_fs_rename_test.ts [hunk 1]
```diff
@@ -1,34 +0,0 @@
...
-Deno.test({
-  name: "ASYNC: renaming a file",
-  async fn() {
-    const file = Deno.makeTempFileSync();
-    const newPath = join(parse(file).dir, `${parse(file).base}_renamed`);
-    await new Promise((resolve, reject) => {
-      rename(file, newPath, (err) => {
-        if (err) reject(err);
-      });
-    })
```
Comment (lucacasonato) [inline std/node/_fs/_fs_rename_test.ts:15]: @AliBasicCoder This one never resolves.

Comment (AliBasicCoder) [reply std/node/_fs/_fs_rename_test.ts:15]: @lucacasonato thank you

**Comment by AliBasicCoder** [comment]
see #7666

**Comment by AliBasicCoder** [comment]
see #7921

**Pull request merged**